NX-493: Metadata Service

cstamas opened the pull request:
Metadata service encapsulating:
- NX-493: metadata store based on OrientDB
- parsing (using low-memory Jackson Streaming API) of JSON documents
- producing streamed content (for downstream consumption)
- covering NX-498 hosted ("private registry") and NX-611 proxy ("registry proxy") and NX-612 group ("aggregated registry") use case for now
- NX-491 Use of GSON dropped

cstamas committed:
NX-493: First draft of the service using orientdb

cstamas committed:
Smaller fixes

cstamas committed:
NX-493: Proxy and hosted services, a lot of simplifaction

Paved the way for more services. Also, removed a lot of redundant
ifaces proved to be unneded. Finally, tests (that should be remain
as is since it uses _real registry_!) covering hosted and
proxy use cases.

cstamas committed:
NX-493: Attachment handling and leaner parsing

cstamas committed:
NX-493: Files that were left out from prev commit

cstamas committed:
NX-493: Separated out generator from producer

To be able to get properly post-processed packages but not
in "raw" (content locator ) form

cstamas committed:
NX-493: Added tarball url filtering

cstamas committed:
NX-493: NPE guard

cstamas committed:
NX-493: Removed unused method for hosted reposes

jdillon commented on nexus-npm-repository-plugin/src/main/java/com/bolyuba/nexus/plugin/npm/metadata/internal/orient/EntityHandler.java:
```diff
@@ -0,0 +1,36 @@
+import com.orientechnologies.orient.core.record.impl.ODocument;
+
+/**
+ * Created by cstamas on 29/07/14.
```
please fix default javadoc junk

cstamas committed:
NX-493: Integrated service with proxy and hosted repo
NX-611: NPM Proxy repository
NX-498: NPM Hosted repository
NX-490: Registry root metadata is not cached (WIP)
NX-491: Use of GSON dropped

Initial implementation of proxy and hosted repositories
using the new metadata service. Dropping many of the (now)
unused stuff, also, GSon proves to be not needed anymore.

cstamas committed:
NX-493: Javadoc

cstamas committed:
NX-493: Multiple improvements

Changes:
* obey "localOnly" to fix UI problem on NPM proxy config (that it goes to fetch all)
* fix registry URI building
* proxy now supports "expire", "nfc" and real remote control
* support for NPM CLI's "/-/all" special API
* improved logging

cstamas committed:
NX-493: De-mavenize

By mistake, some repo helper classes were extending
maven specific helpers instead of the core ones. Nothing
biggy, but this is not needed.

cstamas committed:
NX-493: Refactored internals and groups

cstamas committed:
NX-493: More refactoring

Paving the way to simplify group support

cstamas committed:
NX-493: Cleanup

cstamas committed:
NX-493: Group NPM repository

cstamas committed:
NX-493: Cleanup and group support

Current NX2.x UI is somewhat limited with custom
repo types, hence removed repo type and instead using
the core ones (as any other plugin does) as UI cannot
create custom repo types for groups. Same done for
hosted and proxy repo.

cstamas committed:
NX-493: Disable redeploy for hosted repositories

Obey configuration, and prevent redeploys if needed.

cstamas committed:
NX-493: Fixed registry root caching

And made tests not go remote to fetch it now, as it takes ages
and registry.npmjs.org might get angry on us :D

cstamas committed:
NX-493: Remove orient shutdown hook

That results in exception and NX shut down

cstamas committed:
NX-493: Fix exception on startup

Caused by the fact that getResultingActionOnWrite
received path that is not coming from a package request

cstamas committed:
NX-493: Attempt to improve registry root download

As it seems registry is dropping connection on us, or somewhere
some resource is mishandled

cstamas committed:
NX-493: Attachments were lost on update

cstamas committed:
NX-493: Proxy logic was flawed and not robust

The root document update was flawed, happened always
except when needed.

The root processing is now caged, the exception
will not bubble up but error will be reported,
salvaging as much as possible. User still can force
re-download by expiring caches from root.

cstamas committed:
NX-493: Fix root generation
